File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minimumCost(std::vector<int>& cost) {
        // Sort costs in descending order
        std::sort(cost.rbegin(), cost.rend());
        
        int total_cost = 0;
        for (int i = 0; i < cost.size(); ++i) {
            // Every 3rd item (0-indexed 2, 5, 8) is free
            if (i % 3 != 2) {
                total_cost += cost[i];
            }
        }
        
        return total_cost;
    }
};

File: 2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.java
import java.util.Arrays;

class Solution {
    public int minimumCost(int[] cost) {
        // Sort the array in ascending order
        Arrays.sort(cost);
        
        int totalCost = 0;
        int n = cost.length;
        
        // Iterate backwards from the most expensive candy
        for (int i = n - 1; i >= 0; i--) {
            // We pay for the 1st and 2nd candies in every group of 3
            // The 3rd candy (where (n - i) is a multiple of 3) is free
            if ((n - i) % 3 != 0) {
                totalCost += cost[i];
            }
        }
        
        return totalCost;
    }
}